src/detect-ssh-proto-version.c

```diff
@@ -286,12 +286,10 @@ static int DetectSshVersionTestParse01 (void)
 {
     DetectSshVersionData *ssh = NULL;
     ssh = DetectSshVersionParse(NULL, "1.0");
-    if (ssh != NULL && strncmp((char *) ssh->ver, "1.0", 3) == 0) {
-        DetectSshVersionFree(NULL, ssh);
-        return 1;
-    }
+    FAIL_IF_NOT(ssh != NULL && strncmp((char *)ssh->ver, "1.0", 3) == 0);
```

Comment:
Tip: Read out what the statement is trying to do and then convert them as is using FAIL/PASS API.

1. Let's break these conditions down for better readability. Unless needed, make such statements separate in different FAIL_IF.. calls.
2. Try to avoid double negation and keep the intention/readability of the statement clear.
3. There are more available FAIL_* macros in https://github.com/OISF/suricata/blob/master/src/util-unittest.h#L89 that may be better suited to your work here. 😉

Reply:
Perfect @inashivb! I will do that! Thank you for the tips